Add remaining pills mechanism

teodorciuraru opened the pull request:
**Summary**: Adds a mechanism through which the users will have to keep track of their medicine in order to use the application and will also help them set reminders for when they remain out of medicine.

**User Profile View Controller**

A new class that, for the moment, implements only the pill taking mechanism. It allows the user to set a _Reminder Time_ for a specific interval of not having enough pills.

**Example**: I want to be reminded when I lack pills for more than **4 days**.

Moreover, this class is the one responsible with letting the user change his current medicine stock. The user will type in their current available pills and the stock will automatically update (and also the rest of the app's screens).

A label will be presented on the bottom, stating for how long the user still has pills. **If the value is lower than the value he set a reminder to track, the label will turn red and a local notification will appear.**.

**Use-case**:

The volunteers sets their current medicine stock in the _User Profile_ tab and can start selecting if they did or didn't take their pill from the home screen or from the calendar. 

In the calendar, the user still have the possibility to change decisions made for dates in the past (change _Take_ pill into _Didn't take_), **without the need to posses any medicine**, because decisions before the _Refill Date_ (the last date he modified his remaining medicine stock) won't affect the current stock.

When the user changes a decision made **after the refill date**, she will either take her medicine back (if setting _I didn't took my pill_ when she said she took) or the medicine stock will be decremented when confirming pill taking.

**Changes**
- _WidgetHandler_ logic was modified to support this mechanism.
- A MedicineStockManager was created in order to add the logic behind tracking if the user has enough pills and when to decrement / increment stock.
- All classes now call the MedicineStockManager's _addRegistry()_ instead of the RegistriesManager one, in order for the manager to confirm if the user can add a registry at that moment (if only he has enough pills).
- A _UserProfile_ class was created where, for the moment, only the pill taking UI was implemented.
- Adds new _Unit tests_ for the MedicineStockManager.
- A _MedicineNotificationsHelper_ was added in order to add a new set of custom notifications when the user lacks enough pills (based on his reminder).
- The _NotificationManager_ now archives the notifications persistently, in order to remove correctly when needed.

Commit:
Add remaining pills mechanism.

bphenriques commented:
Teodor, please provide a more detailed public description of the changes you made (widget, bugfix log, etc) so that this pull request is well documented.

I would ask you to submit smaller pull requests in the future, it's easier for me, for you and for anyone curious who wants to read this pull request.

bphenriques commented on malaria-ios/Managers/MedicineStockManager.swift:
```diff
@@ -0,0 +1,58 @@
+      let registryAdded = medicine.registriesManager.addRegistry(date, tookMedicine: tookMedicine, modifyEntry: modifyEntry)
+      
+      // Only update stock if the entry was added successfully
+      if registryAdded.0 && registryAdded.noOtherEntryFound {
```
What represents 0 ? I assume this is a tuple. Replace by proper identifiers to be clear what is happening.

e.g.
var person = (firstName: "John", lastName: "Smith")

person.firstName instead of person.0

bphenriques commented on malaria-ios/Managers/NotificationManager.swift:
```diff
@@ -3,52 +3,76 @@ import UIKit
+  
+  /// Unschedule all notifications with the category
+  public func unsheduleNotification(){
+    //        for event in UIApplication.sharedApplication().scheduledLocalNotifications!
```
Remove

teo029 replied:
Why should I remove the scheduleNotification and unscheduleNotification?

teo029 replied:
You probably only mean the array.

bphenriques commented on malaria-ios/ViewControllers/WidgetSettingsManager.swift:
```diff
@@ -0,0 +1,77 @@
+      NSUserDefaults(suiteName: Constants.Widget.AppGroupBundleID)!.setBool(value, forKey: self.rawValue)
+    }
+    
+    /// Gets the value of the boolean user setting. If the value isn't set, sets it as default value and returns
```
update comments, isn't this widget settings?

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+  
+  func getRegisteredMedicine() {
+    medicine = []
+    medicine.append(medicineManager!.getCurrentMedicine()!)
```
why not assign directly [ (...).getCurrentMedicine()! ] ?

teodorciuraru replied:
Nice one!

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+    medicine.append(medicineManager!.getCurrentMedicine()!)
+    
+    // To get all medicine use only:
+    // medicine = medicineManager!.getRegisteredMedicines()
```
The code is clear on this. I understand what you did here. I suggest remove this comment and document on the official documentation

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+    
+    shouldPresentNotification ? psnm!.scheduleNotification(NSDate()) : psnm!.unsheduleNotification()
+    
+    let defaultBrownTint = self.remindMeWeeksButton.titleLabel?.textColor
```
I don't understand, can't this be set in the storyboard?

teodorciuraru replied:
It will be rewritten in the newest patch.

teodorciuraru replied:
Imo, this method does too many things. I've divided it into multiple methods in this commit that will come.

bphenriques replied:
Sure. I will re-review the PR after the following commit.

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+  }
+  
+  @IBAction func remindMeWeeksPressed(sender: UIButton) {
+    let alertController = UIAlertController(title: "Set reminder for:", message: nil, preferredStyle: UIAlertControllerStyle.Alert)
```
Move "magic" strings and numbers (this case just string) to a dedicated constant similar to: 
https://github.com/systers/malaria-app-ios/blob/develop/malaria-ios/ViewControllers/PlanTrip/PlanTripViewController.swift#L305

In the future we might need to Localize the strings therefore moving the strings from the UIAlertController is a necessary step.

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+    
+    cell.updateCellWithParameters(self,
+                                  name: name,
+                                  remainingMedicine: String(remainingMedicine),
```
Pass the integer and let the cell process the required transformation to a string and not the view controller itself

teodorciuraru replied:
Ok. But why?

bphenriques replied:
Due to reason described above. For it makes more sense the cell converting the integer to cell as required. Imagine if the cell would make arithmetics with the value? If you pass as a string, the cell would have to parse it to int... its preferable to pass the values as they are.

bphenriques commented on malaria-ios/ViewControllers/UserProfileViewController.swift:
```diff
@@ -0,0 +1,199 @@
+  // http://stackoverflow.com/questions/24977047/tableview-cell-on-ipad-refusing-to-accept-clear-color
+  
+  func tableView(tableView: UITableView, willDisplayCell cell: UITableViewCell, forRowAtIndexPath indexPath: NSIndexPath) {
+    
```
remove line

teo029 replied:
Which line? Do I not know how to read comments? :))

I see multiple lines marked. Probably the link.

teo029 replied:
Oh, you probably mean the last line indicated in the comment. Ok.

bphenriques commented on malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift:
```diff
@@ -0,0 +1,45 @@
+                                remainingMedicine: String,
+                                indexPath: NSIndexPath) {
+    pillName.text = name
+    print(remainingMedicine)
```
Remove. If you ever want to log something, use the logger :)

bphenriques commented on malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift:
```diff
@@ -0,0 +1,45 @@
+    quantityTextField.text = remainingMedicine
+    quantityTextField.delegate = self
+    quantityTextField.tag = indexPath.row
+    toolBar = ToolbarWithDone(viewsWithToolbar: [quantityTextField])
```
I don't think it is easy but try to just set this value once (you are always creating a new object whenever the cell is requested to be rendered)

bphenriques commented on malaria-ios/ViewControllers/Constants.swift:
```diff
@@ -1,9 +1,26 @@
+    
+    // We found a problem that didn't allow us to click on the table view's cell if we only set the cell
+    // height (60) and not add ~30-50 offset to it
+    static let CellHeightAndOffset: CGFloat = 60 + 30
```
Put this constant on the view controller itself

bphenriques commented on malaria-ios/Managers/NotificationManager.swift:
```diff
@@ -3,52 +3,76 @@ import UIKit
+    
+    // Save the notifications persistently
+    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
+    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
```
Use the UserSettingsManager which is a wrapper for standardUserDefaults. Is **cleaner** and removes the boiler plate code and that constant

teodorciuraru replied:
The UserSettingsManager only adds Strings and Bools. Should I add a method to add objects? (the local notification array)

bphenriques replied:
Yes

bphenriques replied:
You'll need to add unit tests for that ;)

There is a file already but methods to test custom objects not.

teodorciuraru replied:
What should I test? If the NSUserDefault saves my value correctly?

bphenriques replied:
Of course! And other tests you think you are relevant that covers most
cases.

2016-06-25 21:11 GMT+01:00 Teodor Ciuraru notifications@github.com:

> In malaria-ios/Managers/NotificationManager.swift
> https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493299:
> 
> > -  /// - parameter `NSDate`:: fireTime
> > -  /// - returns: `UILocalNotification`: local notification
> > -  private func createNotification(fireDate: NSDate) -> UILocalNotification{
> > -    let localNotification = UILocalNotification()
> > -    localNotification.fireDate = fireDate
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName;
> > -    localNotification.alertAction = alertAction
> > -    localNotification.alertBody = alertBody
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName
> > -    localNotification.category = category
> >   +
> > -    scheduledNotifications.append(localNotification)
> >   +
> > -    // Save the notifications persistently
> > -    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
> > -    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
> 
> What should I test? If the NSUserDefault saves my value correctly?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493299,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEgjsZGOjDqK1XRKsnzDtg7iKI_8zsKhks5qPYttgaJpZM4I-I45
> .

## 

Os melhores cumprimentos,
Bruno Henriques

teodorciuraru replied:
Sorry, I don't understand. Apple guarantees the fact that NSUserDefaults saves NSObjects that can be later retrieved. Do you have any special case in mind?

bphenriques replied:
The wrapper is a abstraction of the NSUserDefaults. NSUserDefaults is
tested (as you said) but the wrapper isn't and you are using the later. I
understand that it might be redundant in your opinion but it really isn't !
;) Moreover, you are storing custom objects like notifications, a
interesting test would be adding a object with custom fields and if the
fields after loading remain the same.

2016-06-25 21:15 GMT+01:00 Teodor Ciuraru notifications@github.com:

> In malaria-ios/Managers/NotificationManager.swift
> https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493345:
> 
> > -  /// - parameter `NSDate`:: fireTime
> > -  /// - returns: `UILocalNotification`: local notification
> > -  private func createNotification(fireDate: NSDate) -> UILocalNotification{
> > -    let localNotification = UILocalNotification()
> > -    localNotification.fireDate = fireDate
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName;
> > -    localNotification.alertAction = alertAction
> > -    localNotification.alertBody = alertBody
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName
> > -    localNotification.category = category
> >   +
> > -    scheduledNotifications.append(localNotification)
> >   +
> > -    // Save the notifications persistently
> > -    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
> > -    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
> 
> Sorry, I don't understand. Apple guarantees the fact that NSUserDefaults
> saves NSObjects that can be later retrieved. Do you have any special case
> in mind?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493345,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEgjsZ1npQnuuIioBAUz8lGZiGFug_eoks5qPYxlgaJpZM4I-I45
> .

## 

Os melhores cumprimentos,
Bruno Henriques

bphenriques replied:
Another example... imagine if NSUserDefaults changes internally so that
custom objects are only stored withing specific conditions. With those unit
tests, you will be able to tackle the issue more rapidly.

2016-06-25 21:15 GMT+01:00 Teodor Ciuraru notifications@github.com:

> In malaria-ios/Managers/NotificationManager.swift
> https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493345:
> 
> > -  /// - parameter `NSDate`:: fireTime
> > -  /// - returns: `UILocalNotification`: local notification
> > -  private func createNotification(fireDate: NSDate) -> UILocalNotification{
> > -    let localNotification = UILocalNotification()
> > -    localNotification.fireDate = fireDate
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName;
> > -    localNotification.alertAction = alertAction
> > -    localNotification.alertBody = alertBody
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName
> > -    localNotification.category = category
> >   +
> > -    scheduledNotifications.append(localNotification)
> >   +
> > -    // Save the notifications persistently
> > -    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
> > -    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
> 
> Sorry, I don't understand. Apple guarantees the fact that NSUserDefaults
> saves NSObjects that can be later retrieved. Do you have any special case
> in mind?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493345,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEgjsZ1npQnuuIioBAUz8lGZiGFug_eoks5qPYxlgaJpZM4I-I45
> .

## 

Os melhores cumprimentos,
Bruno Henriques

teodorciuraru replied:
Got it!

bphenriques replied:
Your question is similar to should we do unit tests for everything? IMO, as
much as possible to assure that your app is running correctly. This is even
more useful after code rewrites or when updating the dependencies to newer
versions. Moreover and last reason, developing unit tests for your module
equals to wrapping it up and gives confidence to your deliverable.

2016-06-25 21:15 GMT+01:00 Teodor Ciuraru notifications@github.com:

> In malaria-ios/Managers/NotificationManager.swift
> https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493345:
> 
> > -  /// - parameter `NSDate`:: fireTime
> > -  /// - returns: `UILocalNotification`: local notification
> > -  private func createNotification(fireDate: NSDate) -> UILocalNotification{
> > -    let localNotification = UILocalNotification()
> > -    localNotification.fireDate = fireDate
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName;
> > -    localNotification.alertAction = alertAction
> > -    localNotification.alertBody = alertBody
> > -    localNotification.soundName = UILocalNotificationDefaultSoundName
> > -    localNotification.category = category
> >   +
> > -    scheduledNotifications.append(localNotification)
> >   +
> > -    // Save the notifications persistently
> > -    let localNotificationArrayData = NSKeyedArchiver.archivedDataWithRootObject(scheduledNotifications)
> > -    NSUserDefaults.standardUserDefaults().setObject(localNotificationArrayData, forKey: Constants.localNotificationsArray)
> 
> Sorry, I don't understand. Apple guarantees the fact that NSUserDefaults
> saves NSObjects that can be later retrieved. Do you have any special case
> in mind?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493345,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEgjsZ1npQnuuIioBAUz8lGZiGFug_eoks5qPYxlgaJpZM4I-I45
> .

## 

Os melhores cumprimentos,
Bruno Henriques

bphenriques commented on malaria-ios/ViewControllers/Constants.swift:
```diff
@@ -1,9 +1,26 @@
+  }
+  
+  struct UserProfile {
+    static let CellReuseIdentifier = "User Profile Pill Cell Identifier"
```
All the UI constants, unless they are used elsewhere should be on the ViewController itself (this case).

bphenriques commented on malaria-ios/ViewControllers/Constants.swift:
```diff
@@ -1,9 +1,26 @@
+  
+  struct UserProfile {
+    static let CellReuseIdentifier = "User Profile Pill Cell Identifier"
+    static let PillReminderValue = "Pill Reminder Value"
```
Are you using this variable? I think not as you are using in the UserSettings enum (which is great).

If you are not using, remove.

bphenriques commented on malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift:
```diff
@@ -0,0 +1,45 @@
+  
+  var toolBar: ToolbarWithDone!
+  
+  func updateCellWithParameters(delegate: SaveRemainingPillsProtocol,
```
"WithParamenters" is redundant.

Weird indentation, is your screen lacking width? Personally, I would put everything on the same line since it improves redability and this method doesn't have that many parameters that would justify multiple lines.

teodorciuraru replied:
I'm trying to keep lines under 80 characters, but I'm not sure if it's necessary anymore (I will check the coding styles again).

teodorciuraru replied:
I find it easier to read on multiple lines. Here's a debate:

http://stackoverflow.com/questions/110928/is-there-a-valid-reason-for-enforcing-a-maximum-width-of-80-characters-in-a-code

bphenriques replied:
I understand your concern. I usually put between 80 and 100ish. But I don't
follow it religiously... depends on the case.

2016-06-25 20:52 GMT+01:00 Teodor Ciuraru notifications@github.com:

> In malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift
> https://github.com/systers/malaria-app-ios/pull/52#discussion_r68493112:
> 
> > @@ -0,0 +1,45 @@
> > +import UIKit
> > +import DoneToolbarSwift
> > +
> > +// MARK: - Table View Cell
> > +
> > +class UserProfilePillTableViewCell: UITableViewCell {
> > +
> > -  @IBOutlet weak var pillName: UILabel!
> > -  @IBOutlet weak var quantityTextField: UITextField!
> >   +
> > -  weak var delegate : SaveRemainingPillsProtocol?
> >   +
> > -  var toolBar: ToolbarWithDone!
> >   +
> > -  func updateCellWithParameters(delegate: SaveRemainingPillsProtocol,
> 
> I'm trying to keep lines under 80 characters, but I'm not sure if it's
> necessary anymore (I will check the coding styles again).
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/systers/malaria-app-ios/pull/52/files/1bd98847e91e9af59711c1205e2643b2daf28fd7#r68493112,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEgjsT0Ts5tB_yKzdps-7-51lYkV7r9kks5qPYbwgaJpZM4I-I45
> .

## 

Os melhores cumprimentos,
Bruno Henriques

bphenriques commented on malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift:
```diff
@@ -0,0 +1,45 @@
+extension UserProfilePillTableViewCell: UITextFieldDelegate {
+  
+  func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
+    return string.isNumber()
```
Nice ;)

bphenriques commented:
@teo029 Please fix (or discuss with me) the proposed resolutions. I will still analyse it with greater detail this evening. In short, the code looks great however, when you use tuples, please identify what is each position explicitly. You scattered (..).0 on the user tests and it is still not clear to be what represents that value which a identifier would solve. IMO Constants related with view controller should remain in the view controller since they are not used anywhere else and will only impact the view controller it self. When makes sense, use @IBInspectable on those constants. All the remaining we have been discussing.

You told me that you have a new commit that fixes some bugs, just push and the pull-request will be updated automatically.

Commit:
Fix Mefloquine bug, add Notification Manager unit tests
- Fixed mefloquine bug regarding not calculating when to show notification correctly.
- Restructured UserProfileVC.
- Added first version of the User Profile icon.
- Added unit tests for the notification manager to check if it saves the notifications locally.

bphenriques commented:
"[!] Your Podfile has had smart quotes sanitised. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice."

Remove the smart quote (avoid using textedit that automatically does that).

bphenriques commented on Podfile:
```diff
@@ -22,4 +22,8 @@ end
     pods
+end
+
+target 'malaria-iosTests’ do
```
here.

bphenriques commented on malaria-iosTests/TestMedicineStockManager.swift:
```diff
@@ -0,0 +1,134 @@
+      return
+    }
+    
+    msm.addRegistry(NSDate().dateByAddingTimeInterval(secondsTillYesterday), tookMedicine: true)
```
use the existing extensions:
NSDate() + secondsTillYesterday.second

You have to use int but in your case it suits.

bphenriques commented on malaria-iosTests/TestMedicineStockManager.swift:
```diff
@@ -0,0 +1,134 @@
+    XCTAssertEqual(medicine.remainingMedicine, 1)
+  }
+  
+  func testUserHasEnoughPillsToPressYes() {
```
what happens when you press No? Do you have a test for it?

bphenriques replied:
Add tests when tookMedicine is set to false. As expected, the medicineStock should remain.

bphenriques commented on malaria-iosTests/TestMedicineStockManager.swift:
```diff
@@ -0,0 +1,134 @@
+    }
+    
+    medicine.remainingMedicine = 1
+    medicine.lastStockRefill = NSDate().dateByAddingTimeInterval(twoDaysAgo)
```
Use existing NSDate extensions as well.

You can add/subtract 1.day, 1.month, 1.year, etc.

bphenriques commented on malaria-iosTests/TestMedicineStockManager.swift:
```diff
@@ -0,0 +1,134 @@
+    msm.addRegistry(NSDate().dateByAddingTimeInterval(secondsTillYesterday), tookMedicine: true)
+    
+    XCTAssertFalse(medicine.remainingMedicine == 1)
+    XCTAssertTrue(medicine.remainingMedicine == 0)
```
Redudant Assert (same field). Keep one of them.

bphenriques commented on malaria-ios/ViewControllers/UserProfilePillTableViewCell.swift:
```diff
@@ -0,0 +1,48 @@
+  
+  var toolBar: ToolbarWithDone!
+  
+  override func layoutSubviews() {
```
Why layoutSubviews? I believe this is called when it is being rendered therefore, the toolbar would be instantiated here as well continuously. Try to use another approach.

Update, it hsa to be called in layoutSubviews, check if the toolbar is not nil before creating another instance of the toolbar and setting the remaining values

bphenriques commented:
I am having a error when clicking the user profile:

cell.UpdateCell(...) on indexPath. EXC_BAD_ACCESS

Log:

```
2016-06-26 11:09:14.302 malaria-ios[3122:31814] [Crashlytics] Version 3.7.2 (112)
[WARN] - UserSettingsManager::getBool::41   ClearMedicineHistory isn't set, setting and returning default value false
[WARN] - UserSettingsManager::getBool::41   ClearTripHistory isn't set, setting and returning default value false
[INFO] - DidTakePillsViewController::refreshScreen()::97    Refreshing TOOK PILL
[INFO] - DidTakePillsViewController::refreshScreen()::97    Refreshing TOOK PILL
[WARN] - UserSettingsManager::getBool::41   DidConfiguredMedicine isn't set, setting and returning default value false
[INFO] - DidTakePillsViewController::refreshScreen()::97    Refreshing TOOK PILL
[ERRO] - MedicineManager::setCurrentPill::69    No current pill found!
[INFO] - MedicineManager::setCurrentPill::72    Setting Doxycycline as default
[WARN] - MedicineStockManager::hasEnoughPills()::60     User doesn't have enough pills to add an entry.
[INFO] - DidTakePillsViewController::refreshScreen()::97    Refreshing TOOK PILL
[INFO] - DidTakePillsViewController::refreshScreen()::125   No information
[WARN] - MedicineStockManager::hasEnoughPills()::60     User doesn't have enough pills to add an entry.
[WARN] - UserSettingsManager::getBool::41   MedicineReminderSwitch isn't set, setting and returning default value true
[WARN] - UserSettingsManager::getLocalNotifications::86     SaveLocalNotifications isn't set, setting and returning default value <>
2016-06-26 11:09:36.069 malaria-ios[3122:31814] *** -[NSKeyedUnarchiver initForReadingWithData:]: data is empty; did you forget to send -finishEncoding to the NSKeyedArchiver?
[INFO] - NotificationManager::scheduleNotification::27  Sheduling PILL_REMINDER to 26-June-2016 11:09
[INFO] - DidTakePillsViewController::refreshScreen()::97    Refreshing TOOK PILL
[INFO] - DidTakePillsViewController::refreshScreen()::125   No information
[WARN] - MedicineStockManager::hasEnoughPills()::60     User doesn't have enough pills to add an entry.
[WARN] - UserSettingsManager::getString::64     PillReminderValue isn't set, setting and returning default value 
[INFO] - NotificationManager::scheduleNotification::27  Sheduling PillStatusReminder to 26-June-2016 11:09
(lldb) 
```

bphenriques commented:
Now that I notice, you a have this error:

```
2016-06-26 11:09:36.069 malaria-ios[3122:31814] *** -[NSKeyedUnarchiver initForReadingWithData:]: data is empty; did you forget to send -finishEncoding to the NSKeyedArchiver?
```

I will resume the testing until I the current issues are solved

bphenriques commented:
The icon on iPhone 6s plus is distorted horizontally.

Commit:
Add new User Profile Icon, new MedicineStockManager unit test

bphenriques commented:
The icon looks great now. 

I would just add:
1. Fix the NSDate extensions: NSDate() + 1.day NSDate() - 1.day
2. Remove unused variables (in unit tests)
3. Add green view behind the "1 week" and change the color of textField to green as well to conform to the general theme.

bphenriques commented on malaria-ios/ViewControllers/SupportingFiles/UserSettingsManager.swift:
```diff
@@ -64,7 +66,29 @@ public class UserSettingsManager{
+      }
+      
+      /// Gets the local notifications data for the key. If it is not set, returns the default value.
+      /// - parameter `String: optional`: default value when the variable isn't set, currently, empty data.
```
update comment

Commit:
Add minor refactoring and design fixes

Commit:
Add Peace Corps logo, change UI colors

Commit:
Add new Scroll View constraints in UserProfile
